dev-games/openscenegraph-openmw/openscenegraph-openmw-3.4_p20200425-r100.ebuild

```
@@ -0,0 +1,171 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+LUA_COMPAT=( lua5-{1..2} )
+WX_GTK_VER="3.0-gtk3"
+inherit cmake flag-o-matic lua-single wxwidgets
+
+DESCRIPTION="OpenMW-specific fork of OpenSceneGraph"
+HOMEPAGE="https://github.com/OpenMW/osg"
+MY_COMMIT="8b07809fa674ecffe77338aaea2e223b3aadff0e"
+SRC_URI="https://github.com/OpenMW/osg/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="wxWinLL-3 LGPL-2.1"
+SLOT="0/132" # NOTE: CHECK WHEN BUMPING! Subslot is SOVERSION
+KEYWORDS="~amd64 ~x86"
+IUSE="curl debug doc examples egl ffmpeg fltk fox gdal gif glut gstreamer jpeg
+	lua openexr openinventor osgapps pdf png qt5 sdl sdl2 svg tiff
+	truetype vnc wxwidgets xine xrandr zlib"
+
+REQUIRED_USE="
+	lua? ( ${LUA_REQUIRED_USE} )
+	sdl2? ( sdl )
+	openexr? ( zlib )
+"
+
+# TODO: COLLADA, FBX, GTA, OpenVRML, Performer, DCMTK
+RDEPEND="
+	!dev-games/openscenegraph
+	media-libs/mesa[egl?]
+	virtual/glu
+	virtual/opengl
+	x11-libs/libSM
+	x11-libs/libXext
+	curl? ( net-misc/curl )
+	examples? (
+		fltk? ( x11-libs/fltk:1[opengl] )
+		fox? ( x11-libs/fox:1.6[opengl] )
+		glut? ( media-libs/freeglut )
+		sdl2? ( media-libs/libsdl2 )
+		wxwidgets? ( x11-libs/wxGTK:${WX_GTK_VER}[opengl,X] )
+	)
+	ffmpeg? ( media-video/ffmpeg:0= )
+	gdal? ( sci-libs/gdal:= )
+	gif? ( media-libs/giflib:= )
+	gstreamer? (
+		media-libs/gstreamer:1.0
+		media-libs/gst-plugins-base:1.0
+	)
+	jpeg? ( virtual/jpeg:0 )
+	lua? ( ${LUA_DEPS} )
```

```suggestion
	${LUA_DEPS}
```

I **think** the correct way to use this is to just specify `${LUA_DEPS}` in a *DEPEND block. Since now you can enable `lua_single_target_lua5-1 lua_single_target_lua5-2` and disable `lua`, and maybe the other way around, and not sure what that does to the ebuild. Just depending on `${LUA_DEPS}` should enable `lua` USE flag and enable the specifics correctly.

Let me know if it works fine in your opinion like this.

---

Then it will depend on dev-lang/lua:5.1 (or 5.2, depending on which LUA_SINGLE_TARGET are enabled) even if lua USE flag is disabled.